### PR TITLE
Fixes Assertion failed 'firstILBlock->hasProfileWeight()

### PR DESCRIPTION
**Pull request by briansull**
When creating a new top block we need to use inheritWeight to properly initialize the the block weight and the profile data flag

### CODE REVIEW

**Comment by BruceForstall** [comment]
Fixes #14550

**Comment by BruceForstall** [comment]
@dotnet-bot test Windows_NT x86_arm_altjit Checked corefx_jitstress1
@dotnet-bot test Windows_NT x64_arm64_altjit Checked corefx_jitstress1

**Comment by briansull** [comment]
@BruceForstall  PTAL
 @sandreenko PTAL

File: src/jit/optimizer.cpp [hunk 1]
```diff
@@ -2851,6 +2852,23 @@ bool Compiler::optCanonicalizeLoop(unsigned char loopInd)
         JITDUMP("in optCanonicalizeLoop: redirect top predecessor BB%02u to BB%02u\n", topPredBlock->bbNum,
                 newT->bbNum);
         optRedirectBlock(topPredBlock, blockMap);
+
+        // When we have profile data then the 'newT' block will inherit topPredBlock profile weight
+        if (topPredBlock->hasProfileWeight())
+        {
+            // This corrects an issue when the topPredBlock has a profile based weight
+            //
+            JITDUMP(", hasProfileWeight() returns true");
```
[the remark this answers is not in the document]
Comment (briansull) [reply src/jit/optimizer.cpp]: I will fix

**Comment by briansull** [comment]
@dotnet-bot test Windows_NT x86_arm_altjit Checked corefx_jitstress1
@dotnet-bot test Windows_NT x64_arm64_altjit Checked corefx_jitstress1

**Comment by briansull** [comment]
@dotnet-bot test Windows_NT x86_arm_altjit Checked corefx_jitstress1
@dotnet-bot test Windows_NT x64_arm64_altjit Checked corefx_jitstress1

**Comment by briansull** [comment]
@dotnet-bot test Windows_NT x86_arm_altjit Checked corefx_jitstress1
@dotnet-bot test Windows_NT x64_arm64_altjit Checked corefx_jitstress1

**Comment by briansull** [comment]
@dotnet-bot Test Windows_NT x86 full_opt ryujit CoreCLR Perf Tests Correctness 
@dotnet-bot Test Windows_NT x64 full_opt ryujit CoreCLR Perf Tests Correctness

**Comment by briansull** [comment]
@dotnet-bot Test Windows_NT x86 full_opt ryujit CoreCLR Perf Tests Correctness
@dotnet-bot Test Windows_NT x64 full_opt ryujit CoreCLR Perf Tests Correctness

**Comment by briansull** [comment]
@dotnet-bot test Windows_NT x86_arm_altjit Checked corefx_jitstress1
@dotnet-bot test Windows_NT x64_arm64_altjit Checked corefx_jitstress1

**Pull request merged**